"Use of future reserved word in strict mode" false positive with eslint-env es6

ESLint version: `0.17.1` (and 67936e0d571c613ac54f4097b4235e6411af0b79 gives the same results)

`foo.js`:
```js
/*eslint-env es6 */
let x = 42;
```

`.eslintrc`:
```json
{
    "ecmaFeatures": { "modules": true }
}
```

The false positive:
```
$ ./node_modules/.bin/eslint foo.js 

foo.js
  2:1  error  Use of future reserved word in strict mode

✖ 1 problem (1 error, 0 warnings)
```

--- a/lib/eslint.js
+++ b/lib/eslint.js
@@ -302,11 +302,10 @@ function modifyConfigsFromComments(ast, config, reportingConfig, messages) {
         }
     });
 
-    // apply environment rules before user rules
+    // apply environment configs
     Object.keys(commentConfig.env).forEach(function (name) {
-        var environmentRules = environments[name] && environments[name].rules;
-        if (commentConfig.env[name] && environmentRules) {
-            assign(commentConfig.rules, environmentRules);
+        if (environments[name]) {
+            util.mergeConfigs(commentConfig, environments[name]);
         }
     });
     assign(commentConfig.rules, commentRules);

--- a/tests/lib/eslint.js
+++ b/tests/lib/eslint.js
@@ -1251,9 +1251,8 @@ describe("eslint", function() {
     });
 
     describe("when evaluating code containing /*eslint-env */ block", function() {
-        var code = "/*eslint-env node*/ function f() {} /*eslint-env browser, foo*/";
-
         it("variables should be available in global scope", function() {
+            var code = "/*eslint-env node*/ function f() {} /*eslint-env browser, foo*/";
             var config = { rules: {} };
 
             eslint.reset();
@@ -1266,6 +1265,15 @@ describe("eslint", function() {
                 assert.equal(window.writeable, false);
             });
             eslint.verify(code, config, filename, true);
+        });
+        it("should error on node specific rule", function() {
+            var code = "/*eslint-env node*/ var appHeader = new require('app-header');";
+            var config = { rules: {} };
+
+            eslint.reset();
+            var messages = eslint.verify(code, config, filename, true);
+            assert.equal(messages.length, 1);
+            assert.equal(messages[0].ruleId, "no-new-require");
         });
     });
 

